org.rdkit.knime.types/python3/knime/types/ext/rdkit.py

```diff
@@ -0,0 +1,214 @@
+    # Any attempt to work with this data will fail.
+    class Chem:
+        class Mol:
+            pass
+
+    class ChemicalReaction:
+        pass
+
+    class ExplicitBitVect:
+        pass
+
+    class UIntSparseIntVect:
+        pass
+
+    class IntSparseIntVect:
+        pass
+
+    class LongSparseIntVect:
+        pass
+
+    class ULongSparseIntVect:
+        pass
+
+
+class RDKitMolAdapterValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, Chem.Mol)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+
+        value = Chem.Mol(storage["0"])
+        return value
+
+    def encode(self, value):
+        if value is None:
+            return None
+        return {"0": value.ToBinary(), "1": None}
+
+
+class RDKitMolValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, Chem.Mol)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+        return Chem.Mol(storage)
+
+    def encode(self, value):
+        if value is None:
+            return None
+        return value.ToBinary()
+
+
+class RDKitReactionValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, ChemicalReaction)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+        return ChemicalReaction(storage)
+
+    def encode(self, value):
+        if value is None:
+            return None
+        return value.ToBinary()
+
+
+class RDKitFingerprintValueFactory(kt.PythonValueFactory):
+    def __init__(self):
+        kt.PythonValueFactory.__init__(self, ExplicitBitVect)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+        import rdkit
+
+        length = int.from_bytes(storage[:8], byteorder="little")
+        fp = rdkit.DataStructs.CreateFromBinaryText(storage[8:])
+        return fp
+
+    def encode(self, value):
+        if value is None:
+            return None
+        import rdkit
+
+        length_bytes = len(value).to_bytes(length=8, byteorder="little")
+        return length_bytes + rdkit.DataStructs.BitVectToBinaryText(value)
+
+
+class AbstractRDKitCountFingerprintValueFactory(kt.PythonValueFactory):
+    def __init__(self, dtype):
+        self._dtype = dtype
+        kt.PythonValueFactory.__init__(self, dtype)
+
+    def decode(self, storage):
+        if storage is None:
+            return None
+
+        # construct a new count fingerprint of type _dtype and fill it
+        sparse_vec = self._dtype(len(storage))
+        for i, v in enumerate(storage):
+            if v:
+                sparse_vec[i] = v
+        return sparse_vec
+
+    def encode(self, value):
+        if value is None:
+            return None
+
+        ba = bytearray(value.GetLength())
+        for idx, v in value.GetNonzeroElements().items():
+            ba[idx] = v
```

Comment:
This will throw a `ValueError` when a count exceeds 255.
This almost never happens with normal molecules, but it certainly is possible.
I'd suggest:
```suggestion
            ba[idx] = v % 256
```